api/src/main/java/io/strimzi/api/kafka/model/KafkaSpec.java

```diff
@@ -39,6 +41,7 @@
     private CertificateAuthority clusterCa;
 
     private CertificateAuthority clientsCa;
+    private MaintenanceSpec maintenance;
```

Comment:
Since we add the generic `maintenance` field ... do you expect other fields to be added later as well?

Reply:
Ok, fine with me.

Reply:
I disagree here. Unless we can point to something specific which we expect to add in the future I think we should just have a `Kafka.spec.maintenanceWindows` property. With the approach of "Oh, well we might need something in the future" then we end up with a very deep hierarchy, and end up needing a CRD class for every little thing. If you look at the k8s schemas they have sibling properties like `image` and `imagePullPolicy` and that works fine.

If you're convinced that the approach you're taking here is the right one, I'm not going to argue, but I think we should try to resist defaulting to always adding a new class to hold new properties.

Reply:
I could agree with you because I don't have a real use case for now but ... does it really hurt having a hierarchy here? I know that K8S has `image` and `imagePullPolicy` maybe they didn't think about that?
In our case, for example, we started having `topicOperator` under the `Kafka` resource and then after adding the UO we decided to add an `entityOperator` and having TO and UO inside it instead of adding the `userOperator` just at the same level of the EO. We ended up with adding a new parent field (the `entityOperator`). Just genuine question, is there any technical problem or reason you see around CRD (or something) that justify that having a `maintenance` field, and other fields, under it is not the right way to go?